src/vs/base/common/stream.ts

```diff
@@ -186,7 +186,7 @@ export interface ITransformer<Original, Transformed> {
 	error?: IErrorTransformer;
 }
 
-export function newWriteableStream<T>(reducer: IReducer<T>, options?: WriteableStreamOptions): WriteableStream<T> {
+export function newWriteableStream<T>(reducer: IReducer<T> | null, options?: WriteableStreamOptions): WriteableStream<T> {
```

Comment:
Used `null` on purpose to make it explicit to prevent subtle silent errors. It is too easy to overlook a variable or a function return value being an undefined and pass it over, hence I wanted to make sure the folks that using this to be very specific.

I do appreciate the work that TS has done with `void` and not treating the `foo?: string` and `foo: undefined | string` the same anymore, but there is still the cases when requiring the explicit `null` saves some hair on your head 🤗 Do you think it would be fine with `undefined`? You have more context so happy to revert if you think it won't be a pain in the future.

Reply:
I don't think an explicit `undefined` is that different, but you can use what you want